packages/query-core/src/queryClient.ts

```diff
@@ -585,14 +585,17 @@ export class QueryClient {
 
   getMutationDefaults(
     mutationKey: MutationKey,
-  ): MutationObserverOptions<any, any, any, any> {
+  ): OmitKeyof<MutationObserverOptions<any, any, any, any>, 'mutationKey'> {
     const defaults = [...this.#mutationDefaults.values()]
 
-    let result: MutationObserverOptions<any, any, any, any> = {}
+    const result: OmitKeyof<
+      MutationObserverOptions<any, any, any, any>,
+      'mutationKey'
+    > = {}
 
     defaults.forEach((queryDefault) => {
       if (partialMatchKey(mutationKey, queryDefault.mutationKey)) {
-        result = { ...result, ...queryDefault.defaultOptions }
+        Object.assign(result, queryDefault.defaultOptions)
```

Comment:
> Aligning implementation with getQueryDefaults by using Object.assign

@Nick-Lucas do you remember why you changed this to `Object.assign` in `getQueryDefaults` ?

Reply:
Yes, it's potentially a faster operation and in some cases looks cleaner. That's it. But the difference in speed is probably virtually irrelevant as both approaches are very fast

https://www.measurethat.net/Benchmarks/Show/2485/0/spread-vs-objectassign#:~:text=Executions%20per%20second-,Object.assign%20two%20objects,226555.7%20Ops/sec,-Object.assign%20three